pyrobosim/pyrobosim/utils/general.py

```diff
@@ -99,3 +101,30 @@ def replace_special_yaml_tokens(in_text, root_dir=None):
     out_text = out_text.replace("$PWD", root_dir)
 
     return out_text
+
+
+def parse_color(color):
+    """
+    Parses a color input and returns an RGB tuple.
+
+    :param color: Input color as a list, string, or hexadecimal.
+    :type color: list[float] | str
+    :return: RGB tuple in range (0.0, 1.0).
+    :rtype: list[float]
```

Comment:
instead of casting as a list, why not just return a tuple? The first sentence of the docstring says tuple, anyhow.

Reply:
I initially used a list because the test cases for `robot.py` and `yaml_utils.py` expected the return type to be a list. However, I have now updated the return type to a `tuple` as per the docstring and adjusted the test cases and code accordingly. I believe I’ve covered everything, but please let me know if I missed anything.